src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,157 @@
+package duke;
+
+import java.util.ArrayList;
+
+/**
+ * A class containing the list of tasks
+ * added by the user.
+ *
+ * @author Elbert Benedict
+ */
+public class TaskList {
+    private ArrayList<Task> taskList;
+
+    /**
+     * Contructs a new instance of Tasklist.
+     * The taskList is initially empty.
+     */
+    public TaskList() {
+        taskList = new ArrayList<Task>();
+    }
+
+    /**
+     * Returns the number of tasks.
+     *
+     * @return the number of tasks.
+     */
+    public int getSize() {
+        return taskList.size();
+    }
+
+    /**
+     * Adds a new task to the list of tasks.
+     *
+     * @param task the task to be added.
+     */
+    public void add(Task task) {
+        taskList.add(task);
+    }
+
+    /**
+     * Deletes a task from the list of tasks and
+     * Returns the deleted task message.
+     *
+     * @param index the index of the tasks to be deleted.
+     * @return the deleted task message.
+     * @throws DukeException If index is out of bounds.
+     */
+    public String delete(int index) throws DukeException {
+        //Index out of bounds
+        if (index > taskList.size() || index < 1) {
+            throw new DukeException("Index Is Not Valid");

```

Comment:
"Index Is Not Valid" can be saved as a constant too as it is used multiple times.